Dialog/index.d.ts

```diff
@@ -0,0 +1,47 @@
+import MaterialComponent from '../MaterialComponent';
+import { VNode, PreactHTMLAttributes } from 'preact';
+import Button from '../Button';
+import { MDCFoundation, MDCComponent } from '../MaterialComponentsWeb';
+import { Omit } from '../libs';
+
+declare interface IDialogProps extends JSX.HTMLAttributes {
+  onAccept?: () => void;
+  onCancel?: () => void;
+}
+export default class Dialog extends MaterialComponent<IDialogProps, {}> {
+  static Header: typeof Header;
+  static Body: typeof Body;
+  static Footer: typeof Footer;
+  static FooterButton: typeof FooterButton;
+
+  MDComponent: MDCDialog;
+}
+
+declare class Header extends MaterialComponent<JSX.HTMLAttributes, {}> {}
+
+declare interface IBodyProps extends JSX.HTMLAttributes {
+  scrollable?: boolean;
+}
+declare class Body extends MaterialComponent<IBodyProps, {}> {}
+
+declare class Footer extends MaterialComponent<JSX.HTMLAttributes, {}> {}
+
+declare interface IFooterButtonProps extends Omit<JSX.HTMLAttributes, 'accept'> {
+  accept?: boolean;
+  cancel?: boolean;
+}
+declare class FooterButton extends Button<IFooterButtonProps, {}> {}
```

Comment:
This is still not working because it extends Button, which also inherits `accept: string` from HTMLAttributes (`Type '{ accept: true; }' is not assignable to type 'IFooterButtonProps & IButtonProps & IMaterialComponentProps & ComponentProps<FooterButton>'.`).

Maybe should change back to the way you had before where props interfaces don't extend anything and use `& HTMLAttributes` in the class definition? That way could extend ButtonProps here without inheriting anything other than that?

Reply:
I've moved `JSX.HTMLAttributes` to be the default `PropsType` for `Button`. This will make it possible to override it in `Dialog/index.d.ts`.